tools/third_party/pytest/src/_pytest/assertion/rewrite.py

```diff
@@ -27,7 +27,7 @@
 import py
 
 from _pytest._io.saferepr import saferepr
-from _pytest._version import version
+#from _pytest._version import version
```

Comment:
I wonder if these problems are because our `.tox` version of pytest would now be out of sync with the third_party version.

Could you try updating the pytest versions in `tools/wpt/tox.ini`, `tools/wptrunner/tox.ini`, `tools/wave/tox.ini`, and `tools/tox.ini` to be the same as the one you vendored in?

Reply:
After changing the `rewrite.py ` back and `tools/wpt/tox.ini`, `tools/wptrunner/tox.ini`, `tools/wave/tox.ini`, and `tools/tox.ini` versions, run 
`python3 wpt  run --log-mach=- --channel=dev chrome webdriver/tests/websockets/connect.py`.
I still got
` 0:07.60 INFO STDERR:   File "/usr/local/google/home/shengfa/Workspace/wpt/tools/third_party/pytest/src/_pytest/assertion/rewrite.py", line 30, in <module>
 0:07.60 INFO STDERR:     
 0:07.60 INFO STDERR: from _pytest._version import version
 0:07.60 INFO STDERR: ModuleNotFoundError
 0:07.60 INFO STDERR: : 
 0:07.60 INFO STDERR: No module named '_pytest._version'
`

Reply:
There is an import error from `tools/third_party/pytest/src/_pytest/__init__.py` to import `from ._version import version as __version__` but `rewrite.py` is still trying to use the value directly.

Reply:
I think it's probably some path or install issue. In `tools/third_party/pytest/setup.cfg` it has
`[check-manifest]
ignore =
    src/_pytest/_version.py
`
so I think that file should somehow be generated for the import to succeed but I am not sure how to achieve that.

Reply:
Ok, so yes this is because `pytest` now uses a setup that builds the `_version.py` file. We should probably be grabbing the built pytest from pip directly, e.g. https://pypi.org/project/pytest/6.1.1/#files

I tried downloading the `.tar.gz`, unzipping it, and copying it into our vendored `pytest` directory. I think executed `git add -f tools/third_party/pytest` and got:

```
$ git status
HEAD detached from shengfa/websockets
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        deleted:    tools/third_party/pytest/.coveragerc
        deleted:    tools/third_party/pytest/.gitattributes
        deleted:    tools/third_party/pytest/.gitblameignore
        deleted:    tools/third_party/pytest/.github/FUNDING.yml
        deleted:    tools/third_party/pytest/.github/ISSUE_TEMPLATE/1_bug_report.md
        deleted:    tools/third_party/pytest/.github/ISSUE_TEMPLATE/2_feature_request.md
        deleted:    tools/third_party/pytest/.github/ISSUE_TEMPLATE/config.yml
        deleted:    tools/third_party/pytest/.github/PULL_REQUEST_TEMPLATE.md
        deleted:    tools/third_party/pytest/.github/config.yml
        deleted:    tools/third_party/pytest/.github/labels.toml
        deleted:    tools/third_party/pytest/.github/workflows/main.yml
        deleted:    tools/third_party/pytest/.github/workflows/release-on-comment.yml
        deleted:    tools/third_party/pytest/.gitignore
        deleted:    tools/third_party/pytest/.pre-commit-config.yaml
        deleted:    tools/third_party/pytest/.readthedocs.yml
        deleted:    tools/third_party/pytest/.travis.yml
        new file:   tools/third_party/pytest/PKG-INFO
        modified:   tools/third_party/pytest/setup.cfg
        new file:   tools/third_party/pytest/src/_pytest/_version.py
        new file:   tools/third_party/pytest/src/pytest.egg-info/PKG-INFO
        new file:   tools/third_party/pytest/src/pytest.egg-info/SOURCES.txt
        new file:   tools/third_party/pytest/src/pytest.egg-info/dependency_links.txt
        new file:   tools/third_party/pytest/src/pytest.egg-info/entry_points.txt
        new file:   tools/third_party/pytest/src/pytest.egg-info/not-zip-safe
        new file:   tools/third_party/pytest/src/pytest.egg-info/requires.txt
        new file:   tools/third_party/pytest/src/pytest.egg-info/top_level.txt
```

That seems like a sensible diff to me.

However when I `wpt run` I get `iniconfig` errors from `findpaths.py` in pytest - did you run into that?

```
 0:08.73 INFO STDERR: Exception in thread 
 0:08.73 INFO STDERR: Thread-1
 0:08.73 INFO STDERR: :
 0:08.73 INFO STDERR: Traceback (most recent call last):
 0:08.73 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/wptrunner/wptrunner/executors/base.py", line 668, in _run
 0:08.73 INFO STDERR:     
 0:08.74 INFO STDERR: self.result = True, self.func(self.session, self.path, self.timeout)
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/wptrunner/wptrunner/executors/base.py", line 630, in do_wdspec
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: return pytestrunner.run(path,
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/wptrunner/wptrunner/executors/pytestrunner/runner.py", line 42, in run
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: do_delayed_imports()
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/wptrunner/wptrunner/executors/pytestrunner/runner.py", line 24, in do_delayed_imports
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: import pytest
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/third_party/pytest/src/pytest/__init__.py", line 5, in <module>
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: from _pytest.assertion import register_assert_rewrite
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/third_party/pytest/src/_pytest/assertion/__init__.py", line 8, in <module>
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: from _pytest.assertion import rewrite
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/third_party/pytest/src/_pytest/assertion/rewrite.py", line 37, in <module>
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: from _pytest.config import Config
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/third_party/pytest/src/_pytest/config/__init__.py", line 42, in <module>
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: from .findpaths import determine_setup
 0:08.74 INFO STDERR:   File "/home/smcgruer/github/wpt/tools/third_party/pytest/src/_pytest/config/findpaths.py", line 10, in <module>
 0:08.74 INFO STDERR:     
 0:08.74 INFO STDERR: import iniconfig
 0:08.74 INFO STDERR: ModuleNotFoundError
 0:08.74 INFO STDERR: : 
 0:08.74 INFO STDERR: No module named 'iniconfig'
```

Reply:
Thanks! Updated pytest with
> I tried downloading the .tar.gz, unzipping it, and copying it into our vendored pytest directory. I think executed git add -f tools/third_party/pytest

and no longer need to change version manually.

> However when I wpt run I get iniconfig errors from findpaths.py in pytest - did you run into that?

This should not happen as iniconfig is added in tools/third_party and localpath in one of the commits.